api/app/utils/daily_pdf_gen.py

```diff
@@ -0,0 +1,43 @@
+from typing import List, Mapping
+import pdfkit
+from jinja2 import Template
+
+from app.schemas.hfi_calc import FireCentre, HFIResultResponse, PlanningArea, WeatherStation
+from app.utils.pdf_data_formatter import response_2_daily_jinja_format
+
+daily_template_path = "api/app/utils/daily_template.html"
+daily_rendered_path = "api/app/utils/daily_rendered.html"
+
+output_file_path = "api/app/utils/out.pdf"
+
+
+def generate_daily_pdf(result: HFIResultResponse, fire_centres: List[FireCentre]):
+    """Generate a daily PDF"""
+    # Shift hydrated fire centres into dicts keyed by ids
+    fire_centre_dict: Mapping[int, FireCentre] = dict()
+    planning_area_dict: Mapping[int, PlanningArea] = dict()
+    station_dict: Mapping[int, WeatherStation] = dict()
+    for fire_centre in fire_centres:
+        fire_centre_dict[fire_centre.id] = fire_centre
+        for planning_area in fire_centre.planning_areas:
+            planning_area_dict[planning_area.id] = planning_area
+            for station in planning_area.stations:
+                station_dict[station.code] = station
+
+    fire_centre_name = fire_centre_dict[result.selected_fire_center_id].name
+    with open(daily_template_path, 'r') as daily_template, open(daily_rendered_path, 'w') as new_page:
```

Comment:
Same as above - you won't have access - you need a temporary or in memory file.

Reply:
In memory now, as of: https://github.com/bcgov/wps/pull/1763/commits/4e333fd132ab77cf21de524d5282b9bd8f573941